server/router/api.go

```diff
@@ -137,6 +142,15 @@ func apiRoutes(e *gin.RouterGroup) {
 					repo.PATCH("/cron/:cron", session.MustPush, api.PatchCron)
 					repo.DELETE("/cron/:cron", session.MustPush, api.DeleteCron)
 
+					agentBase := repo.Group("/agents")
```

Comment:
Do we really need repo agents? As you can apply filters to org / (user org) agents for specific repos I don't think we need them TBH.

Reply:
it does not add more complexity and allow a even more granular management ...
... one of the things i want to do next is to have a prioritized queue and with such it should priorize the lowest hirarchy always if it can ...

beside that, you could be in an org and be an repo admin but not an org admin ... this would cover the case

Reply:
> it does not add more complexity and allow a even more granular management ...

It does add more code and thereby complexity ... 😉 As the benefit isn't granted / rather unclear atm we should not add them for now IMO

Reply:
still even in terms of UX it's better to have the option to let users just attach agents to repos.

And one of the next things I'll imediatly work on in terms of features (before the the compile step) will be to have a priorized queue, so that if an repo agent exist, it will always be prefered over org or instance wide agents. does this do it?

Reply:
move conversation to main chat: https://github.com/woodpecker-ci/woodpecker/pull/3539#issuecomment-2366869300